### ISSUE TITLE
Bug: [prefer-spread] false-positive with `this`

### ISSUE THREAD

**Issue by regseb** [eslint/eslint#16704]
### Environment

Node version: v18.12.1
npm version: 9.2.0
Local ESLint version: v8.30.0
Global ESLint version: N/A
Operating System: Ubuntu 22.04.1 LTS

### What parser are you using?

Default (Espree)

### What did you do?

[**ESLint Playground**](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IHByZWZlci1zcHJlYWQ6IFtcImVycm9yXCJdICovXG5cbmNsYXNzIEZvbyB7XG4gICNmbiA9ICgpID0+IHt9O1xuICAjYXJncyA9IFtcImJhclwiXTtcblxuICBjb25zdHJ1Y3RvcigpIHtcbiAgICBjb25zdCB0aGF0ID0gdGhpcztcbiAgICBjb25zdCBmbiA9IHRoaXMuI2ZuO1xuICAgIGNvbnN0IGFyZ3MgPSB0aGlzLiNhcmdzO1xuXG4gICAgdGhpcy4jZm4uYXBwbHkodGhpcywgdGhpcy4jYXJncyk7IC8vIHByZWZlci1zcHJlYWQhXG4gICAgdGhpcy4jZm4uYXBwbHkodGhpcywgYXJncyk7IC8vIHByZWZlci1zcHJlYWQhXG5cbiAgICB0aGlzLiNmbi5hcHBseSh0aGF0LCB0aGlzLiNhcmdzKTtcbiAgICBmbi5hcHBseSh0aGlzLCB0aGlzLiNhcmdzKTtcbiAgICB0aGlzLiNmbi5hcHBseShcImJhelwiLCB0aGlzLiNhcmdzKTtcbiAgICAoKCkgPT4ge30pLmFwcGx5KHRoaXMsIHRoaXMuI2FyZ3MpO1xuICAgIHRoaXMuI2ZuLmFwcGx5KHRoaXMsIFtcImJhelwiXSk7XG4gIH1cbn1cbiIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6ImxhdGVzdCIsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=):

```js
/* eslint prefer-spread: ["error"] */

class Foo {
  #fn = () => {};
  #args = ["bar"];

  constructor() {
    const that = this;
    const fn = this.#fn;
    const args = this.#args;

    this.#fn.apply(this, this.#args); // prefer-spread!
    this.#fn.apply(this, args); // prefer-spread!

    this.#fn.apply(that, this.#args);
    fn.apply(this, this.#args);
    this.#fn.apply("baz", this.#args);
    (() => {}).apply(this, this.#args);
    this.#fn.apply(this, ["baz"]);
  }
}
```

### What did you expect to happen?

Report no errors.

### What actually happened?

ESLint reports two errors:

> 12:5 &nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Use the spread operator instead of '.apply()'. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp; ([prefer-spread](https://eslint.org/docs/rules/prefer-spread))
> 13:5 &nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Use the spread operator instead of '.apply()'. &nbsp;&nbsp;&nbsp;&nbsp;&nbsp; ([prefer-spread](https://eslint.org/docs/rules/prefer-spread))

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by regseb** [comment]
Oops. I was wrong, `prefer-spread` works. The mistake is mine.
